test/e2e/adminapi_cluster_update.go

```diff
@@ -0,0 +1,52 @@
+package e2e
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	"github.com/Azure/ARO-RP/pkg/api/admin"
+)
+
+var _ = Describe("[Admin API] Cluster admin update action", func() {
+	BeforeEach(skipIfNotInDevelopmentEnv)
+
+	It("should be able to run cluster update operation on a cluster", func() {
+		var oc *admin.OpenShiftCluster = &admin.OpenShiftCluster{}
+		ctx := context.Background()
+		resourceID := resourceIDFromEnv()
+
+		By("triggering the update via RP admin API")
+		resp, err := adminRequest(ctx, http.MethodPatch, resourceID, nil, json.RawMessage("{}"), oc)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(resp.StatusCode).To(Equal(http.StatusOK))
+
+		By("checking provisioning state")
+		Expect(oc.Properties.ProvisioningState).To(Equal(admin.ProvisioningStateAdminUpdating))
+		Expect(oc.Properties.LastProvisioningState).To(Equal(admin.ProvisioningStateSucceeded))
+
+		By("waiting for the update to complete")
+		err = wait.PollImmediate(10*time.Second, 30*time.Minute, func() (bool, error) {
+			oc = getCluster(ctx, resourceID)
+			return oc.Properties.ProvisioningState == admin.ProvisioningStateSucceeded, nil
+		})
+		Expect(err).NotTo(HaveOccurred())
+	})
+})
+
+func getCluster(ctx context.Context, resourceID string) *admin.OpenShiftCluster {
+	var oc admin.OpenShiftCluster
+	resp, err := adminRequest(ctx, http.MethodGet, resourceID, nil, nil, &oc)
+	Expect(err).NotTo(HaveOccurred())
+	Expect(resp.StatusCode).To(Equal(http.StatusOK))
+	return &oc
+}
```

Comment:
+1 on this - I think it's likely that this kind of thing will need to grow out in due course into a fuller client which in dev talks to the RP on localhost, and in int/prod calls geneva actions using their API.